src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dnsbl.xml

```diff
@@ -54,21 +54,23 @@
             be blocked. Blocking first-level domains is not supported.
         </help>
     </field>
+    <field>
+        <id>unbound.dnsbl.rcode</id>
+        <label>Return Response Code</label>
+        <type>dropdown</type>
+        <advanced>true</advanced>
+        <help>
+            Use the DNS response code for entries in the blocklist.
+        </help>
+    </field>
     <field>
         <id>unbound.dnsbl.address</id>
         <label>Destination Address</label>
         <type>text</type>
         <advanced>true</advanced>
         <help>
             Destination ip address for entries in the blocklist (leave empty to use default: 0.0.0.0).
-            Not used when "Return NXDOMAIN" is checked.
+            Only used when "Return Response Code" is "A".
         </help>
     </field>
-    <field>
-        <id>unbound.dnsbl.nxdomain</id>
```

Comment:
this needs a model migration (https://docs.opnsense.org/development/frontend/models_migrations.html#migrations). If you need help with that, we have to postpone the merge for a while (when I have some time to test this myself).